core/src/test/java/io/confluent/kafka/schemaregistry/rest/SchemaRegistryConfigTest.java

```diff
@@ -189,4 +195,33 @@ public void defaultMutabilityMode() throws RestConfigException {
     assertEquals(true, config.getBoolean(SchemaRegistryConfig.MODE_MUTABILITY));
   }
 
+  @Test
+  public void testSslConfigOverride() throws RestConfigException, SchemaRegistryException {
+    Properties props = new Properties();
+
+    props.setProperty(RestConfig.LISTENERS_CONFIG, "alice://localhost:123, bob://localhost:456, https://localhost:789");
+    props.setProperty(SchemaRegistryConfig.INTER_INSTANCE_LISTENER_NAME_CONFIG, "alice");
+    props.setProperty(RestConfig.LISTENER_PROTOCOL_MAP_CONFIG, "alice:https, bob:https");
+    props.setProperty(RestConfig.SSL_KEYSTORE_LOCATION_CONFIG, "/mnt/keystore/keystore.jks");
+    props.setProperty("listener.name.alice." + RestConfig.SSL_KEYSTORE_LOCATION_CONFIG , "/mnt/keystore/internal/keystore.jks");
+    SchemaRegistryConfig config = new SchemaRegistryConfig(props);
+
+    NamedURI internalListener = KafkaSchemaRegistry.getInternalListener(config.getListeners(),
+      config.getString(SchemaRegistryConfig.INTER_INSTANCE_LISTENER_NAME_CONFIG),
+      SchemaRegistryConfig.HTTPS);
+    Map<String, Object> overrides = config.getOverriddenSslConfigs(internalListener);
+    assertEquals("/mnt/keystore/internal/keystore.jks", overrides.get(SchemaRegistryConfig.SSL_KEYSTORE_LOCATION_CONFIG));
```

Comment:
Should we also check internalListener.scheme, host, port, etc... configs?

Reply:
There are existing tests for those in KafkaSchemaRegistry, I have changed them as per the new code -  testGetPortForIdentityPrecedence, testGetPortForIdentityNoListeners etc.